### ISSUE TITLE
[Bug] [iOS] GestureRecognizers not working in iOS 13.4

### ISSUE THREAD

**Issue by dan5602** [xamarin/Xamarin.Forms#10162]
### IMPORTANT UPDATE REGARDING THIS ISSUE. PLEASE READ BEFORE COMMENTING

⚠️ Please update to latest stable in Visual Studio for Mac (v8.5.2) and Windows (v16.5.3), we have released the update that includes the fix. Thanks again for your patience.

----

### Description
After upgrading to xcode 11.4 and iOS 13.4 GestureRecognizers (TapGestureRecognizer) have stopped working on iOS. Android works as expected.

### Steps to Reproduce

1. Create a Xamarin Forms mobile application
2. Add a label with a TapGestureRecognizer
3. Try to invoke TapGestureRecognizer

### Expected Behavior
TapGestureRecognizer is invoked

### Actual Behavior
TapGestureRecognizer is not invoked

### Basic Information
When updating MacOS to the below configuration, TapGestureRecognizer stops working in the iOS platform.

### Configuration
Mac Catalina 10.15.4
VS2019 16.5.2
Xamarin VS package 13.6.0.11
Xamarin.Forms 4.5.0.495

### Reproduction Link
[Link](https://1drv.ms/u/s!AuOz60zuc2Drgs9dkkKnX3kV23ZfBA?e=7tbcLU)

**Comment by solomonfried** [comment]
This may be associated to the issue I am seeing with Selections on CollectionView

[https://github.com/xamarin/Xamarin.Forms/issues/10160]

**Comment by lhughey** [comment]
YES. This is a critical error that shouldn't have made it past UI smoke tests. All of my gesturerecognizers in iOS are broken, across multiple projects. 

I'm also using 
Mac Catalina 10.15.4
VS2019 16.5.2
Xamarin VS package 13.6.0.11
Xamarin.Forms 4.5.0.495
Xcode 11.4

**Comment by solomonfried** [comment]
To be fair to Xamarin, the error seems to have been introduced with a new version of MacOS. It would be nice if the Xamarin group were ahead of this and were testing the MacOS releases before they were generally available.

**Comment by thomasgalliker** [comment]
Still, I ask myself (not the first time) what quality checks are done before such a release? Xamarin could (my point of view) say we're not supporting Cataline 10.15.4 and xcode 11.4 until we have a stable release. I'm not sure if they have a real chance to test things properly when I see the rate of updates they're releasing. 

However... Thanks in advance to those who solve this bug 🥇

**Comment by EvanMulawski** [comment]
May be due to this new delegate method on `UIKit.UIGestureRecognizerDelegate`:

[`gestureRecognizer:shouldReceiveEvent:`](https://developer.apple.com/documentation/uikit/uigesturerecognizerdelegate/3538976-gesturerecognizer?language=objc)

**Comment by chamons** [comment]
Appears related to https://github.com/xamarin/xamarin-macios/issues/8255 and @dalexsoto is running it down there.

**Comment by InquisitorJax** [comment]
Pretty easy to repro: just create a new Master Detail from the template, run the iOS app.
None of the items on the main page with navigate through to item detail because of the broken tapgesturerecognizer behavior.

**Comment by dan5602** [comment]
@InquisitorJax - Looks like the issue has been found and fixed already.

**Comment by chamons** [comment]
Fix is in PR, once we get a package we will validate. We believe we have this sucker pinned down.

**Comment by InquisitorJax** [comment]
@chamons that's great news - what does the release eta look like - I would imagine very many people who upgraded are now dead in the water because of this :(

**Comment by rezamohamed** [comment]
I literally wasted so much time with the Mac OS upgrade and Visual Studio not syncing and now for the past day dealing with this thinking it was my code. Pretty frustrated with Xamarin right about now...I am just venting. Sorry!

**Comment by ezavil** [comment]
In a physical device with 13.3 OS it seems to work correctly, in the simulator with version 13.4 it fails...

**Comment by mikeluken** [comment]
I am also dead in the water waiting on this fix.  Do we know an ETA on when the fix will be pushed??

**Comment by chamons** [comment]
I don't have a date to share, but you may have a few options if you are on macOS.

- Downgrading to previous stable (https://download.visualstudio.microsoft.com/download/pr/8f94ca38-039a-4c9f-a51a-a6cb33c76a8c/21e09d8084eb7c15eaa07c970e0eccdc/xamarin.ios-13.14.1.39.pkg)
- We'll have a Xamarin.iOS with the fix out of our build machine in a few hours that you can install. I'll post it here once it is complete.

**Comment by rezamohamed** [comment]
@chamons when you say a fix out, will it be a nuget 4.6.x pre version? Just trying to understand what to look out for.

**Comment by chamons** [comment]
The fix will be in a new Xamarin.iOS SDK itself, not a Xamarin.Forms nuget. There is an interaction between a newly bound iOS API and Xamarin.Forms that is causing the issue at hand.

That is part of the reason that rolling back to older iOS pkg will "fix" this, as I noted [here](https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-607500960).

**Comment by rezamohamed** [comment]
Rolling back wouldn't work if you are developing on Visual Studio Windows and using the Pairing agent to Mac. I get the complaining that there is a Xamarin.iOS mismatch and the pairing agent will upgrade Xamarin.iOS. 
![image](https://user-images.githubusercontent.com/18199852/78190240-45217580-7439-11ea-89a2-7d43fd0ca4d3.png)

**Comment by InquisitorJax** [comment]
I managed to work around it by installing VS 16.6 Preview 2 - app seems to work ok with the alternative X.iOS installation used there - not sure I want to rely on a preview build tho.
Good luck to the devs on this - hope the fix is out soon ;)

**Comment by lhughey** [comment]
As someone else mentioned, download the 13.3 simulators. The 13.3 sims are a 3gb download and is is much faster than trying to rollback VS.

**Comment by rezamohamed** [comment]
@lhughey the rollback is not of VS, its of the Xamari.iOS framework SDK on the Mac machine - it is much faster to do that than a 3gb download of sims :-)

**Comment by mikeluken** [comment]
I see that this ticket is now closed.  How/where do I download Xamarin.iOS SDK update???

**Comment by rezamohamed** [comment]
@mikeluken https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-607500960

**Comment by mikeluken** [comment]
> * We'll have a Xamarin.iOS with the fix out of our build machine in a few hours that you can install. I'll post it here once it is complete.

@chamons - The build looks like it has been complete for a few hours now.  Any ETA on when this will be posted?  Please note that as I am on a Windows machine developing, rolling back to the previous version is not an option.  And I have a deadline to complete this project my the morning.  Until you post the updated SDK, I am at a standstill.  This was a pretty large break in functionality and I really do need the fix posted about 6 hours ago.  Please provide an update ASAP.

**Comment by rezamohamed** [comment]
@mikeluken you shouldn't be at standstill per se, if you are developing for iOS you would need a Macbook to pair to regardless. You can still code in Windows and just test it using your Mac. As far as rolling out to devices, it shouldn't be a problem as it works fine on devices. If you really do want to pair and test, your other option is to use the iOS 13.3 simulators.

**Comment by dalexsoto** [comment]
Hello @mikeluken, the issue got closed by GitHub because of the merge of the commit with the fix, in the meantime you can download the new **Notarized** Xamarin.iOS/Mac pkgs from the GitHub commit status checks of any of our branches (see below image for an example), this particular one lives inside our branch called [d16-5-xcode11.4](https://github.com/xamarin/xamarin-macios/commits/d16-5-xcode11.4)

![image](https://user-images.githubusercontent.com/204671/78204185-ed016800-7466-11ea-82c8-2f02b89a4c33.png)

The packages were created some minutes ago and for your convenience I have pasted the links below:

- [xamarin.ios-13.16.0.13.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/d16-5-xcode11.4/b75deaf8277c77200bdb9349d7e677ab07622fad/17/package/notarized/xamarin.ios-13.16.0.13.pkg)
- [xamarin.mac-6.16.0.13.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/d16-5-xcode11.4/b75deaf8277c77200bdb9349d7e677ab07622fad/17/package/notarized/xamarin.mac-6.16.0.13.pkg)

You can test the above packages and let us know if you encounter any further issues, about the availability via the VS updaters should be coming soon.

**Comment by rezamohamed** [comment]
@dalexsoto the problem with the newer packages is that you still can't use a Windows machine to pair to that Mac because of the Xamarin Mac-Windows mismatch issue right? Even rolling one version down makes pairing the issue, so it needs to be a coordinated update for VS Win and Mac.

**Comment by itoledo** [comment]
> @dalexsoto the problem with the newer packages is that you still can't use a Windows machine to pair to that Mac because of the Xamarin Mac-Windows mismatch issue right? Even rolling one version down makes pairing the issue, so it needs to be a coordinated update for VS Win and Mac.

just cancel when VS asks for an "upgrade" (in this case it'll be a downgrade), you'll get a warning about "problems installing", it will connect anyway.

![image](https://user-images.githubusercontent.com/1848210/78206477-512b2880-7475-11ea-9152-ad04651a219a.png)

**Comment by mikeluken** [comment]
@dalexsoto - How/where do I install the .pkg files?  I know in Visual Studio on the PC.  But not sure how to reference the downloaded file...

**Comment by dalexsoto** [comment]
@mikeluken you should be able to install these pkg files in your mac build host by double clicking on the pkg file and follow the installation prompts

**Comment by Markus2807** [comment]
Hi, even after i installed both of these pgk-Files on my Mac the problem still exists.

VS on Windows is warning me that the versions are not compatible and i should update. If i press cancel it still connects. Thats okay. But if i publish the app to my physical device none of the GestureRecognizers is working.

VS  16.5.2
iPhone Xs 13.4
MacOS: 10.15.4
xamarin.mac 6.16.0.13
xamarin.ios 13.16.013

**Comment by ianthetechie** [comment]
With all due respect, this has to be the worst managed software project of all time. You make high school script kiddies look good. How on earth does this stuff make it past testing and into the stable channel? I've run into at least 10 critical, obvious, totally preventable bugs with Xamarin.Forms, and it's getting really old.

Thanks @chamons for the pkg link. That seems to revert the damage temporarily.

**Comment by samhouts** [comment]
I'll reopen this until it's published for visibility.

**Comment by maksym-husiev** [comment]
So, when we'll have the update? Or I missed something?

**Comment by thedee** [comment]
I have same issue.  I am dead in the water.

**Comment by maksym-husiev** [comment]
I spent several hours finding out why my app not working on iOS... :(

**Comment by rezamohamed** [comment]
you shouldn't be dead in the water. Read https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-607584893

**Comment by EvanMulawski** [comment]
I can confirm that this SDK update package works, even when using VS on Windows. Thanks to the Xamarin team for prioritizing this and getting a fix out quickly.

**Comment by ianvink** [comment]
> I can confirm that this SDK update package works, even when using VS on Windows. Thanks to the Xamarin team for prioritizing this and getting a fix out quickly.

@EvanMulawski  What did you install (Provide a link if you could, that would be muchly appreciated by those Googling)

**Comment by EvanMulawski** [comment]
@ianvink The Xamarin.iOS SDK package linked in https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-607585466

In VS, dismiss the incompatible SDK warning and it will connect anyway.

**Comment by maksym-husiev** [comment]
Also can confirm. Working fine on Mac

**Comment by ndastur** [comment]
@ianthetechie I have to agree. I am fairly sure I speak for a large volume of the Xamarin Forms community when I say we would all rather you guys concentrated on a stable project that got the basic right rather than worrying about non-features like CSS for XAML. Come on this sort of stuff is just awful. And why 2 weeks?

**Comment by solomonfried** [comment]
I did get it to run and build. But now when I try to create a Bundle, VS2019 on Windows cannot find my Provisioning Profile
Any ideas?

**Comment by swansca79** [comment]
@ndastur It's two weeks because they are actually going to test it, and it will take them that long to figure out how to test something for once...

**Comment by solomonfried** [comment]
Regarding the Provisioning Profile mentioned above, I think the issue is that in the Signing Identity dropdown, I previously had an option for iPhone Distribution.  That option is no longer available.

**Comment by seanrodda** [comment]
Has the initial update that broke the gestures been pulled? I use Bitrise for build and deploy, they update each weekend to the latest stable and want to make sure that it won't break.

**Comment by solomonfried** [comment]
I know this is not directly related to the issue. But I finally have the app working but I can't see any Signing Identities (aside from Automatic) in the Bundle Signing dialog.
I have created a Provisioning Profile in the Developer Portal and I can see it in the Apple Account on VS2019 on Windows. But there is no association between the Provisioning Profile and a Signing Identity

**Comment by rscholey** [comment]
Just a small request to the Xamarin guys.  Please stop adding fancy new features that nobody needs, do more testing and clear up some of the 1,000s of issues that are out there.  Oh and please communicate the same message to the VS team as those guys appear to do even less testing than you do :-(

**Comment by Deepfreezed** [comment]
Thanks for working on this.

**Comment by vhugogarcia** [comment]
Thanks for working on this issue! Really appreciate it! 💪

**Comment by ndastur** [comment]
I guess those of us that pay significant subscription fees to Microsoft feel less inclined to say thank you for something that a corporation that makes billions of dollars ought to be able to test and have some governance around. Small businesses really suffer when they can't release updates for two weeks because a fundamental part of a framework fails and has been released as production ready.

**Comment by ndastur** [comment]
I have also noticed that there are no iOS UI tests (only Android / UWP) list in the README. Is that because like the rest of us Microsoft can't get proper iOS UI testing to work in Xamarin Forms?

**Comment by GiampaoloGabba** [comment]
Well, this is huge, I think there are tons of people screaming at their monitor trying to find what broken their app :)
Given that i'm pretty sure we will need to wait at least 2 weeks for the next VS update, maybe you should enhance the visibility of the fixed .pkg files somewhere... I dont know, a blog post on your site, a newsletter... just post the link somewere and dont leave it buried in this issue comments....

I hope i'm wrong (regarding the 2 weeks) cause this is issue is too big to wait 2 weeks to get the fix shipped in visual studio... i mean... come on...

**Comment by matheuscschenfeld** [comment]
I had it yesterday and I solved by downgrading the xamarin.iOS.Framework to version 13.14.1.39 that you can find at https://aka.ms/xvs/pkg/macios/13.14.1.39

Remove the current framework from /Library/Frameworks and install this pkg

**Comment by kevongreen** [comment]
@matheuscschenfeld Thanks for sharing. This was very helpful.

**Comment by tsafadi** [comment]
I am using Syncfusion controls and my ListViews also don't respond anymore to Tap and DoubleTap events. However, if I change the property SelectionGesture to value "Hold" and link the ListView with a ItemHolding event, it works fine. Everyone else experiencing this?

**Comment by PureWeen** [comment]
For anyone running CI builds https://github.com/jonathanpeppers/boots is a great way to install specific iOS packages (like the fixed one). You can use azure pipeline, dotnet, or cake to install whatever package you need to.

**Comment by developer9969** [comment]
Hi all,
can Install this package on windows using boots? I did try and I get an error "file is not a valid VSIX package".
Or is this fix just for mac?
thanks

**Comment by PureWeen** [comment]
The fix is mainly if you have a CI process that needs to prepare builds for the app store. Since you can only use a mac to prepare those builds it'll be useful to be able to update the packages on your CI.

If you're using the mac bridge just installing the pkg onto your mac and then dismissing the bridge dialog should be enough to get you going locally

**Comment by amirvenus** [comment]
I have installed those packages but I hope they don't cause further issues for future upgrades.

**Comment by developer9969** [comment]
@PureWeen  thanks for your reply. Provided we get the official fix in the next 1 week/10 days I will be fine waiting and I will not change the pipeline. 

For debugging I use a Mac mini to run the simulators, If my understanding is correct I need to install the pkg on the mac mini  or wait for the fix . Correct?

thanks

**Comment by PureWeen** [comment]
@amirvenus you'll be fine. Once the package is part of the next VSMAC roll out it'll just install over it.

@developer9969 

> For debugging I use a Mac mini to run the simulators, If my understanding is correct I need to install the pkg on the mac mini or wait for the fix . Correct?

Correct. If you install it on the mac mini you should be good to go

**Comment by pmahend1** [comment]
Is the fix is released in Preview channel? @samhouts @PureWeen

**Comment by mostafasany** [comment]
> Is the fix is released in Preview channel? @samhouts @PureWeen

I tried the prerelase 4.6.0.529-pre3 and still not working :(

**Comment by PureWeen** [comment]
@mostafasany this fix is part of the Xamarin.iOS sdks not the Xamarin.Forms sdks

@pmahend1  the bug doesn't manifest on preview but I'm pretty sure that's because preview hasn't been updated to the xcode 11.4 bits so the issue doesn't happen there.

**Comment by titonton** [comment]
Removing the Caching strategy seems to work for ListViews for ItemTapped events. The gesture Recognizers still don't work for sure.

**Comment by tsafadi** [comment]
So is this issue fixed? And is the update released?

**Comment by rezamohamed** [comment]
@dalexsoto why is this reopened? did anyone find another issue? just asking so that if I run into something I know to be aware. Thanks

**Comment by dalexsoto** [comment]
@rezamohamed No, a push to origin in my fork somehow closed this by mistake, the issue is kept open for visibility and the issue is fixed. The fix is coming ASAP in the new release of VIsual Studio for Mac and Windows. Once the fix is released we will close this issue.

**Note: If you just landed here please refer to the issue description for instructions to mitigate**

**Comment by rezamohamed** [comment]
@dalexsoto a little bit off-topic, but are y'all aware of the version mismatch issues that sometimes happen between VSwin and VSmac even while on stable channels? https://forums.xamarin.com/discussion/comment/406654

**Comment by pmahend1** [comment]
@dalexsoto yes. I had reported it on developer community back in December. They closed it saying it will be like that until early 2020,but seems no fix yet. It dint use to happen when I switch preview on Mac.

**Comment by mshwf** [comment]
@dalexsoto Does ASAP mean tht VS's next release will be behind its planned date because of this critical bug? or this will be on it's set date wich may be up to 2 weeks as 16.5.2 just landed few days?

**Comment by VictorKochetkov** [comment]
I've faced the same issue after Visual Studio for Mac and Xcode update :(

**Comment by BlueFire-LLC** [comment]
Everything including my Syncfusion gauge controls works with the current Preview version.

**Comment by dotMorten** [comment]
For me it's pinch-zoom and drag that stopped working. Tap does work.

Apparently if I build an app with the latest VS or VS4Mac version and deploy to an iOS 13.4 device, our drag and pinch gestures stops working.
Apps built with earlier versions are not affected. We can reproduce with both simulator and device.
I even deployed an app to a 13.3 device, verified it worked, then upgraded the device, and the app stopped responding to these gestures.
However apps built months ago with earlier tooling still works. And Apps deployed to 13.3 or older devices but with latest tooling also works.
This to me indicates that it’s a tooling + OS issue.

To reproduce, build an iOS or Forms app with the following project template:
https://marketplace.visualstudio.com/items?itemName=Esri.ArcGISRuntimeTemplates
Deploy to a 13.3 device or simulator, and notice pinch-zoom and pan works.
Upgrade the device to 13.4, or deploy to a 13.4 device, and notice zoom and pan doesn’t work (double-tap-to-zoom and two-finger-rotate does however work).
Repeat with an older version of VS (or xcode?), and the gestures also works.

We use the UIPinchGestureRecognizer and UIPanGestureRecognizer for this functionality. All the functioning recognizers (ie UIRotationGestureRecognizer & UITapGestureRecognizer) are all declared in essentially the same way as the non-functioning ones.

**Comment by hoa-ho** [comment]
After installing the iOS and Mac packages as [mentioned here](https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-607585466), the listview click event works again. So thanks for that! Another problem I have is that image source is not displaying the image anymore. The 2 packages didn't fix this image issue either. 

When I do this:
<Image Source="url here" />

**Comment by PureWeen** [comment]
@hoa-ho that issue is fixed on XF 4.5

**Comment by hoa-ho** [comment]
@PureWeen I updated VS and XF to the latest versions and it's not working

**Comment by PureWeen** [comment]
@hoa-ho create a new issue with a repro please

**Comment by sreerajPR** [comment]
Still reproducible in following configuration (Latest stable versions):

Mac OS X 10.15.4 macOS Catalina

Visual Studio Community 2019 for Mac
Version 8.5.1 (build 42)
Xamarin.iOS
Version: 13.16.0.11 

XCode Version 11.4 (11E146)
iOS Simulator OS: 13.4

XF Version 4.5.0.530

What is an approximate ETA for a fix of this issue ?

**Comment by stefancs50** [comment]
> 
> 
> Still reproducible in following configuration (Latest stable versions):
> 
> Mac OS X 10.15.4 macOS Catalina
> 
> Visual Studio Community 2019 for Mac
> Version 8.5.1 (build 42)
> Xamarin.iOS
> Version: 13.16.0.11
> 
> XCode Version 11.4 (11E146)
> iOS Simulator OS: 13.4
> 
> XF Version 4.5.0.530
> 
> What is an approximate ETA for a fix of this issue ?

same here too!

**Comment by thisisthekap** [comment]
Situation on our side:

* Current Stable: Broken
* Current Preview: Working

**Comment by seanrodda** [comment]
If you read the comments, a fix has not been released to stable yet. But you can download  the build that has the fix in it, if you need it immediately

**Comment by Colonel666** [comment]
Updating to the latest version (preview) solved the problem.
https://docs.microsoft.com/en-us/visualstudio/mac/install-preview?view=vsmac-2019

**Comment by mshwf** [comment]
Any solution for VS for Windows (stable or preview) or Mac (stable only)?

**Comment by astralmaster** [comment]
How on earth did this get past UI tests before releasing..

**Comment by ndastur** [comment]
@astralmaster from what I can see in the README there are no UI tests at all for iOS.
I think this bug should be a defining moment for Xamarin Forms. I am certain the dev community wants a tested and stable product vs non-features like CSS for XAML. Or Shell.  Look at Flutter. No UI generation yet massive uptake because things like hot reload actually work and work well. The inner development cycle with Xamarin Forms is horribly long.
If I wasn't supporting an app built previously on XF I'm not sure I'd start a project with XF now.

**Comment by zerokewl88** [comment]
Following...

> Updating to the latest version (preview) solved the problem.
> https://docs.microsoft.com/en-us/visualstudio/mac/install-preview?view=vsmac-2019

I can see in your link, your linked the Mac VS Preview, but i just tried this in Windows Preview 16.6.0.2.1 and don't see any changes / fixes in my listview controls being able to select a item. 

Did you test this only in Mac ?

**Comment by SunnyMukherjee** [comment]
FYI, the TapGestureRecognizer does not work in the iOS simulator either on Windows or in the Mac.  It does work on a physical device.

That tells me a discrepancy exists in the configuration between the simulator and an actual device.  If you can move any commands into another control, now would be the time until they get this resolved.  For example, I had gesture recognizers in a CollectionView, but I called my logic instead from the SelectionChangedCommand.  However, I still have a CarouselView that does not support a SelectionChangedCommand yet, so I am stuck using a gesture recognizer on a Grid control.  I tested in the simulator.  It does not work.  I tested on an actual device.  It works.

Someone somewhere in the Xamarin team dropped the ball when testing prior working features in the Xamarin Forms framework against a new version of iOS and Xcode.  Please fix soon because I am in the process of releasing an app.

**Comment by tsafadi** [comment]
@SunnyMukherjee it doesn’t work for me on the physical device either. I uploaded my build to TestFlight and it doesn’t do anything on tap.

**Comment by astralmaster** [comment]
@SunnyMukherjee What iOS version is installed on your physical device? It works for my iPhone 11 with iOS 13.0 but the bug was filed for 13.4. I am unable to test this iOS version on a physical device but it indeed fails in the simulator.

**Comment by gcadmes** [comment]
Fwiw, this is an xcode issue more than it is iOS 13.4.
Meaning, If I compile my app with xcode 11.3.1, the issue goes away (ad-hoc or TestFlight), even for devices running iOS 13.4

**Comment by xamiell** [comment]
Yeah, I have the same problem with a project for my company. After update VS on my MackBook this simple line is not working any more on iOS:

```
<StackLayout.GestureRecognizers>
                 <TapGestureRecognizer Tapped="TapGestureRecognizer_Tapped"/>
</StackLayout.GestureRecognizers>
```

I test this on iOS Emulator but the event is not Invoked. Android works pretty well.

**Comment by astralmaster** [comment]
Confirming that the latest Preview version of Visual Studio for Windows fixes the issue for the Simulator.

**Comment by BitProgress** [comment]
@astralmaster Downloading the newest Visual Studio 2019 for windows (16.6.0 Preview 2.1) works for me too.

**Comment by mshwf** [comment]
A new version of Visual Studio just shipped now, release notes doesn't mention this issue
https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes#16.5.3
So the Xamarin.iOS SDK will be shipped separately?

**Comment by dalexsoto** [comment]
Hello Friends 👋 

Please update to latest stable in Visual Studio for Mac (v8.5.2) and Windows (v16.5.3), we have released the update that includes the fix. Thanks again for your patience and for testing out our candidate builds.

Release notes are yet to be updated but Xamarin.iOS 13.16.0.13 is being included in this release.

**Comment by RodSanford** [comment]
Despite the devs that like to come in here and complain and fling their feces comments around dogging the Xamarin team and their builds, I, for one, appreciate the fact that when they do find an issue in their code, they are quick to respond.  Thank you Sam and team for the quick fix.

**Comment by rezamohamed** [comment]
Is the mono package also supposed to be upgraded?
![image](https://user-images.githubusercontent.com/18199852/78705971-ed20be00-78d3-11ea-9b4f-9fbbae1fefab.png)

**Comment by RodSanford** [comment]
I took the suggested version and it worked.  Go for it.

**Comment by BlueFire-LLC** [comment]
Great job guys. Thanks for the quick response.

**Comment by SunnyMukherjee** [comment]
> @SunnyMukherjee What iOS version is installed on your physical device? It works for my iPhone 11 with iOS 13.0 but the bug was filed for 13.4. I am unable to test this iOS version on a physical device but it indeed fails in the simulator.

I have 13.4 installed on my iPhone.  I tested using 13.4 on the simulator too.  It doesn't work in the simulator but works on the phone.  I will update to the latest version of VS and Xcode tonight and see if it fixes the issue as others have suggested.  I will also update to 13.4.1 on my phone since it was released today.

**Comment by breenbob** [comment]
Great that this was fixed so fast. Has anyone seen an issue around custom renderers not working in 16.5.2, (and now also in 16.5.3 now that I have updated)? They work fine when I use 16.4.5 with a clean bin/obj folders/complete rebuild/reinstall between tests.

**Comment by samhouts** [comment]
@breenbob If that problem persists, please open a new issue here. That sounds new. Thanks!

**Comment by xamiell** [comment]
Yeah, the update works!

**Comment by MujtabaAli** [comment]
Working perfectly, thanks.

**Comment by SunnyMukherjee** [comment]
Thank you to the Xamarin team and the Visual Studio team for the fast turn-around.  This is why I use Xamarin as my cross-platform mobile development framework.

**Comment by HopInTheCloud** [comment]
The update works. Thanks Xamarin Team

**Comment by dotMorten** [comment]
Can also confirm the reproducer I listed above no works, so the ArcGISRuntime Map controls are back in working order

**Comment by taublast** [comment]
Lastest vs 2019 16.5.3 installed xam ios 13.16.0.13 but can't test as provisioning profiles and certificates are not received from mac anymore and are not listed in vs dropdowns on windows

**Comment by InquisitorJax** [comment]
@taublast I think that's a separate issue. I had it with the 16.5.2 update, but this update worked fine. I had to provision new profiles to get it to work again.

**Comment by mshwf** [comment]
@taublast this happens with me a lot. And I have to restart Visual Studio /Mac/ Windows/ re-install the provisioning profiles.. , to get them back in Visual Studio. Sometimes they don't, and I have to build on Mac.

**Comment by myonlinegolfclub** [comment]
The update works. Thanks Xamarin Team

**Comment by CNBoland** [comment]
TapGestureRecognizers are working in my app now as expected.

My setup:
Windows 10 Enterprise 1909
Visual Studio 2019 Professional 16.5.3
macOS Catalina 10.15.4
Xcode 11.4
Visual Studio for Mac Community 8.5.2
Mono Framework MDK 6.8.0.123
Xamarin.iOS 13.16.0.13
iOS (my device) 13.4

**Comment by Waaheeda** [comment]
facing same issue with my setup
Mac OS Version: 10.15.4 (19E266)
Visual Studio Version: VS 2019 Professional  8.5.1 
Xamarin.iOS Version: 13.16.0.11
XCode Version: 11.4

**Comment by Waaheeda** [comment]
And update has fixed issue 

> **thank you Xamarin Team for quick fix**

**Comment by pmahend1** [comment]
is anyone getting below error
```
/Users/Prateek.Mahendrakar/.nuget/packages/xamarin.forms/4.5.0.530/buildTransitive/Xamarin.Forms.targets(4,4): Error MSB4064: The "ValidateOnly" parameter is not supported by the "XamlCTask" task. Verify the parameter exists on the task, and it is a settable public instance property. (MSB4064) (Iris.Mobile.Shared)
```

**Comment by baskren** [comment]
My app just got rejected from the iOS App Store because of this issue.  We need a fix!

**Comment by PureWeen** [comment]
https://github.com/xamarin/Xamarin.Forms/issues/10162#issuecomment-610523506

**Comment by BRICEMARCELKOUADIO** [comment]
Can someone help me please, after the update my vs, it doesn't work yet!

**Comment by zerokewl88** [comment]
> Can someone help me please, after the update my vs, it doesn't work yet!

Delete Obj / Bin Folders, clean, rebuild solution and try closing / opening VS.

What have you tried ?

**Comment by BRICEMARCELKOUADIO** [comment]
> > Can someone help me please, after the update my vs, it doesn't work yet!
> 
> Delete Obj / Bin Folders, clean, rebuild solution and try closing / opening VS.
> 
> What have you tried ?

it works now thanks

**Comment by zerokewl88** [comment]
> > > Can someone help me please, after the update my vs, it doesn't work yet!
> > 
> > 
> > Delete Obj / Bin Folders, clean, rebuild solution and try closing / opening VS.
> > What have you tried ?
> 
> it works now thanks

For reference, and for others that find this thread, what did you do that resolved your problem ?

**Comment by BRICEMARCELKOUADIO** [comment]
> Delete Obj / Bin Folders, clean, rebuild solution and try closing / opening VS.

I just do what you say here, thanks again

**Comment by odapplications** [comment]
Does not work on physical device. Strange how some tap gestures work on ios but not all. All work fine on android.

**Comment by odapplications** [comment]
Ios 13.3 it was not working when set in xaml. But in the .cs file it seems to be working. Using latest xamarin forms.

**Comment by wocar** [comment]
It doesnt work for me, now with iOS 13.5

**Comment by LGFox** [comment]
TapGestureRecognizer doesn't work on iOS 13.5.1 again.

**Comment by gcadmes** [comment]
I'm not having any issues with TapGesture's on iOS 13.5.1

**Comment by andyfmking** [comment]
I'm having issues with multiple tap gesture recognizers on a grid on iOS 13.5

I have a recognizer for 1 tap, and a second recognizer for 2 taps, both on the same grid.

Only gesture recognizer with 1 tap fires on iOS while both gesture for 1 tap, and 2 taps fire on Android.

This means iOS fires only the first gesture with least number of taps and ignores all others.

Expected behavior is as on Android.
When tapped once, handler for 1 tap should fire and when tapped twice the handler for 2 taps should fire e.t.c

Tested on XF 4.6 and 4.7 on Visual studio Mac 8.6.5(build 23) and Microsoft Visual Studio 2019 Ver. 16.6.3

**Comment by OscarHMG** [comment]
I'm using TouchEffect, with panckeview
